File: Server/Game/Entities/GameObject.cs
using System;
using System.Runtime.Serialization;

namespace Server.Game.Entities
{
    [DataContract]
    public abstract class GameObject
    {
        public Guid Id { get; protected set; }
        public Vector Position { get; set; }
        public Vector Direction { get; set; }

        protected GameObject()
        {
            Position = new Vector();
            Direction = new Vector();
        }

        public abstract void Update(long delta);
    }
}

File: Client/src/com/badlogic/core/GameObject.java
package com.badlogic.core;

import com.badlogic.game.GameManager;
import com.badlogic.util.Constants;
import com.badlogic.util.Vector;

import java.awt.*;
import java.awt.image.BufferedImage;
import java.util.UUID;

public abstract class GameObject {
    protected String id = UUID.randomUUID().toString();
    protected Vector position;
    protected GameManager gameManager;
    protected BufferedImage sprite;

    public Vector getPosition() {
        return this.position;
    }

    public String getId() {
        return this.id;
    }

    public abstract void render(Graphics graphics);

    public abstract void update(int delta);
}
